### DISCUSSION TITLE
Timeout not killing long running task

### DISCUSSION THREAD

**Discussion by nerdoza** [community/community#25456] [Actions]
<p>I’ve had several instances now where my test runner fails on a PR and churns through 360 of my precious minutes before it is killed. I have since added <code>timeout-minutes: 10</code> to my job declaration, but it’s continuing to happen (running for the full 360 minutes, rather than 10).  Am I missing something, or is there something I don’t understand about how this works?</p>
<p>Workflow File:</p>
<pre><code class="lang-auto">on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master
    types:
      - opened
      - reopened
      - synchronize

name: Master

jobs:
  test:
    name: Test
    runs-on: ubuntu-latest
    timeout-minutes: 10
    steps:
      - name: Checkout Repository
        uses: actions/checkout@v2

      - name: Use Node.js v12
        uses: actions/setup-node@v1.4.0
        with:
          node-version: 12

      - name: Setup MySQL
        uses: mirromutth/mysql-action@v1.1
        with:
          mysql version: '5.7'
          mysql database: 'test'
          mysql root password: 'password'

      - name: Install
        run: npm ci

      - name: Set ENV
        run: echo "MYSQL_PASS=\"password\"" &gt; .env

      - name: Test
        run: npm test

  release:
    if: github.event_name == 'push' &amp;&amp; github.ref == 'refs/heads/master'
    name: Release
    needs: test
    runs-on: ubuntu-latest
    steps:
      - name: Checkout the repository
        uses: actions/checkout@v2

      - name: Use Node.js v12
        uses: actions/setup-node@v1.4.0
        with:
          node-version: 12

      - name: Check if version has been updated 
        id: check
        uses: EndBug/version-check@v1.2.0
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          diff-search: true

      - name: Push to production
        if: steps.check.outputs.changed == 'true'
        uses: ad-m/github-push-action@master
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          branch: production
          tags: true
          force: true
      
      - name: Create Release
        if: steps.check.outputs.changed == 'true'
        uses: actions/create-release@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          tag_name: v${{ steps.check.outputs.version }}
          release_name: Release v${{ steps.check.outputs.version }}

      - name: Send Notification 
        uses: bayssmekanique/action-simple-slack-notifier@v1
        if: steps.check.outputs.changed == 'true' &amp;&amp; always()
        with:
          token: ${{ secrets.SLACK_BOT_TOKEN }}
          status: ${{ job.status }}
          action: Release
          channel: '#deployment'
          version: ${{ steps.check.outputs.version }}
</code></pre>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/a/b/ab92a7574bb4438f17e18050774db9313e5ce139.png" data-download-href="/uploads/short-url/otNQN9PeVzqG7I1gudcr5uEgmbT.png?dl=1" title="Capture" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/2280384/181091201-898d7f9c-847c-4ce0-bb0c-8a4ce8bceb83.png" alt="Capture" data-base62-sha1="otNQN9PeVzqG7I1gudcr5uEgmbT" width="690" height="474" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/a/b/ab92a7574bb4438f17e18050774db9313e5ce139_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Capture</span><span class="informations">730×502 12.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Simran-B** [comment, marked as answer]
<p>Looks like it’s a single command that runs for hours. Maybe GitHub doesn’t terminate running commands, but only checks if the timeout is exceeded after each command? You should probably open a support ticket.</p>
